### PR TITLE
COM-303 Refactor theming of `SplitButton`

### PR DESCRIPTION
**Pull request by NataliaVizintini**
Added theming for SplitButton, ClassKey and styled elements

### CODE REVIEW

File: packages/admin/admin/src/common/buttons/split/SplitButton.tsx [hunk 1]
```diff
@@ -1,12 +1,22 @@
...
+import { styled, useThemeProps } from "@mui/material/styles";
 import * as React from "react";
 import { PropsWithChildren } from "react";
 
 import { useStoredState } from "../../../hooks/useStoredState";
 import { SplitButtonContext } from "./SplitButtonContext";
 
+export type SplitButtonClassKey = "buttonGroup" | "activeButton" | "popover" | "menuList" | "menuItem";
+
 export interface SplitButtonProps extends ButtonGroupProps<any> {
```
Comment (johnnyomair) [inline packages/admin/admin/src/common/buttons/split/SplitButton.tsx:20]: The slot props are missing here I supposed. They should also be passed to the slots.

File: packages/admin/admin/src/common/buttons/split/SplitButton.tsx [hunk 2]
```diff
@@ -18,19 +28,61 @@ export interface SplitButtonProps extends ButtonGroupProps<any> {
     popoverProps?: Partial<PopoverProps>;
```
Comment (johnnyomair) [inline packages/admin/admin/src/common/buttons/split/SplitButton.tsx:28]: This should probably be deprecated in favor of slot props

File: packages/admin/admin/src/common/buttons/split/SplitButton.tsx [hunk 3]
```diff
@@ -18,19 +28,61 @@ export interface SplitButtonProps extends ButtonGroupProps<any> {
...
 }
 
+const ButtonGroup = styled(MuiButtonGroup, {
```
Comment (johnnyomair) [inline packages/admin/admin/src/common/buttons/split/SplitButton.tsx:31]: Please call this Root and slot `root` to be in line with other components

**Review by jamesricky — CHANGES_REQUESTED** [review]

**Review by jamesricky — APPROVED** [review]

File: packages/admin/admin/src/common/buttons/split/SplitButton.tsx [hunk 4]
```diff
@@ -1,36 +1,95 @@
...
+            menuItem: typeof MuiMenuItem;
+        }> {
     selectIcon?: React.ReactNode;
     selectedIndex?: number;
     onSelectIndex?: (index: number, item: React.ReactElement) => void;
     showSelectButton?: boolean;
     localStorageKey?: string;
     autoClickOnSelect?: boolean;
     storage?: Storage;
-    popoverProps?: Partial<PopoverProps>;
```
Comment (jamesricky) [inline packages/admin/admin/src/common/buttons/split/SplitButton.tsx:18]: Removing this prop is a breaking change. 
Please either add a changeset mentioning `slotProps` should be used instead or keep the prop and mark it as deprecated.

**Review by jamesricky — CHANGES_REQUESTED** [review]

File: packages/admin/admin/src/common/buttons/split/SplitButton.tsx (hunk 4 above)
Comment (NataliaVizintini) [reply packages/admin/admin/src/common/buttons/split/SplitButton.tsx:18]: OK, I marked it as deprecated

Comment (jamesricky) [reply packages/admin/admin/src/common/buttons/split/SplitButton.tsx:18]: Please add the `popoverProps` back to the `Popover` then, in addition to `...slotProps?.popover` so it actually still works.

**Review by jamesricky — DISMISSED** [review]

**Review by johnnyomair — DISMISSED** [review]

**Comment by johnnyomair** [comment]
@NataliaVizintini please fix conflicts. Thanks!

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**